nsxt/resource_nsxt_policy_host_transport_node_collection.go

```diff
@@ -0,0 +1,277 @@
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra/sites/enforcement_points"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+func resourceNsxtPolicyHostTransportNodeCollection() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyHostTransportNodeCollectionCreate,
+		Read:   resourceNsxtPolicyHostTransportNodeCollectionRead,
+		Update: resourceNsxtPolicyHostTransportNodeCollectionUpdate,
+		Delete: resourceNsxtPolicyHostTransportNodeCollectionDelete,
+		Importer: &schema.ResourceImporter{
+			State: nsxtPolicyPathResourceImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"site_path": {
+				Type:         schema.TypeString,
+				Description:  "Path to the site this Transport Zone belongs to",
+				Optional:     true,
+				ForceNew:     true,
+				Default:      defaultInfraSitePath,
+				ValidateFunc: validatePolicyPath(),
+			},
+			"enforcement_point": {
+				Type:        schema.TypeString,
+				Description: "ID of the enforcement point this Transport Zone belongs to",
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
+			},
+			"compute_collection_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Compute collection id",
+			},
+			"sub_cluster_config": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "List of sub-cluster configuration",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"host_switch_config_source": {
+							Type:        schema.TypeList,
+							Required:    true,
+							Description: "List of overridden HostSwitch configuration",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"host_switch_id": {
+										Type:        schema.TypeString,
+										Required:    true,
+										Description: "HostSwitch Id",
+									},
+									"transport_node_profile_sub_config_name": {
+										Type:        schema.TypeString,
+										Required:    true,
+										Description: "Name of the TransportNodeProfile sub configuration to be used",
+									},
+								},
+							},
+						},
+						"sub_cluster_id": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "sub-cluster Id",
+						},
+					},
+				},
+			},
+			"transport_node_profile_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Transport Node Profile Path",
```

Comment:
is this path or id?

Reply:
This is a path - as this is a policy resource, it requires paths as object references. I followed the API naming where this attribute is called `transport_node_profile_id` but maybe it makes more sense to name it `transport_node_profile_path`.